src/supported-operations/thin-edge-supported-operations.md

```diff
@@ -0,0 +1,109 @@
+    * `add` cloud_name operation_name - adds single operation to the list if doesn't exist
+    * `remove` cloud_name operation_name - removes single operation from the list if exists
+    * `list [cloud_name]` - lists all operations, unless specific cloud table name provided, then lists only operations for the cloud if exists
+  * introduction of a concept for parametrized operations - an operation which requires additional configuration or setting (eg c8y log request also requires supported log types)
+    * `add` param cloud_name parameter_name - adds single parameter to the list if doesn't exist
+    * `remove` param cloud_name parameter_name - remove single parameter to the list if doesn't exist
+    * `list` param [cloud_name]` - lists all parameter, unless specific cloud table name provided, then lists only parameters for the cloud if exists
+
+Eg:
+
+* `tedge operations add c8y c8y_Apama`
+* `tedge operations add c8y c8y_BatchAnalytics`
+
+Naming and details subject to change and comments.
+
+> Alternative (1)
+>
+> * use `tedge cli` command to add or remove operations one by one, list all operations, list all operations per cloud
+>   * use tedge subcommand `tedge config`
+>   * add new subcommand for `tedge config` to add remove and list:
+>     * `tedge config add operations c8y c8y_Apama`
+>     * `tedge config remove operations c8y c8y_Apama`
+>   * `tedge config` currently supports following operations:
+>     * `set`
+>     * `unset`
+>     * `list` [cloud_name]
+>
+> Why this approach:
+>
+> * Is consistent with current config API
+>
+> Why not this approach:
+>
+> * May be confusing
+> * `certificate` subcommand also modifies config
+> * `operations` seems to be specialized similarly to `certificate`
+>
+
+## Use in tedge components
+
+As tedge components use tegde configuration to obtain other settings adding new setting for supported operations seems like a logical step to extend it.
+There is no need to add new API for tedge config as it already supports retrieving values from config file.
+
+tedge config may require extension to support arrays of strings to be retrieved, but this can be achieved without much effort.
+
+Eg:
+tedge c8y mapper can easily obtain list of supported operations from the tedge config interface by calling `tedge_config::query(Setting)` interface.
+
+## Persistance
+
+As config is static it will be persistent on the system and will last over reboots and power downs.
+
+## Adding supported operations
+
+Using tedge config to add supported operations allows any tedge components (or even any device system component) to extend the list of supported operations on demand. To add or remove supported operations one can either modify tedge.toml file directly or use tedge cli tool:
+
+`tedge cli` tool can be scripted and therefore when installing new components using tedge software management it could be called as a part of the `finalize` process of the plugin.
```

Comment:
I wanted to tell 2 things at the same time here and it looks like I have told none.

What I meant to say here is that when using software management it can be installed as a part of installation script (eg for apt/deb postinst script may execute it), or if it is a custom plugin supporting other package the finalize phase could invoke some metadata/postintall script. I'll  update it with cleaner thoughts .